### PR TITLE
gh-82180: Update math.factorial(float) doc for Python 3.10

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-82180

----
:books: Documentation preview :books:: https://cpython-previews--105385.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
Update the doc for a change made in Python 3.10: issue #82180, commit 578c3955e0222ec7b3146197467fbb0fcfae12fe.

cc @serhiy-storchaka

File: Doc/library/math.rst [hunk 1]
```diff
@@ -71,8 +71,8 @@ Number-theoretic and representation functions
    Return *n* factorial as an integer.  Raises :exc:`ValueError` if *n* is not integral or
    is negative.
 
-   .. deprecated:: 3.9
-      Accepting floats with integral values (like ``5.0``) is deprecated.
+   .. versionchanged:: 3.10
+      Floating point number (like ``5.0``) is no longer accepted.
```
Comment (serhiy-storchaka) [inline Doc/library/math.rst]: Maybe add "with integral value"?

Comment (vstinner) [reply Doc/library/math.rst]: Sure, done. Would you mind to review the updated PR?

**Review by serhiy-storchaka — APPROVED** [review]
Not sure whether it should be "value" or "values". Was it error in the current writing?

**Comment by vstinner** [comment]
I made the phrasing closer to the previous text. Thanks for reviews @serhiy-storchaka.

> Not sure whether it should be "value" or "values". Was it error in the current writing?

I suppose that it's "values", so I kept the old text ;-)

**Pull request merged**